src/Microsoft.TestPlatform.ObjectModel/Client/Events/DiscoveryCompleteEventArgs.cs

```diff
@@ -38,6 +38,15 @@ public DiscoveryCompleteEventArgs(long totalTests, bool isAborted)
         /// </summary>
         public bool IsAborted { get; private set; }
 
+        // List of sources which were fully discovered
+        public IList<string> FullyDiscoveredSources { get; set; } = new List<string>();
```

Comment:
Also the property should probably be read-only (i.e. setter can probably be omitted).

Reply:
(Although it is surprising that the Metrics property below does have a public setter 😞)